native/cocos/physics/physx/character-controllers/PhysXCharacterController.cpp

```diff
@@ -0,0 +1,274 @@
+    uintptr_t wrapperPtrShape = PhysXWorld::getInstance().getWrapperPtrWithObjectID(hitShape->second);
+    if(wrapperPtrShape == 0){
+        return physx::PxQueryHitType::eNONE;
+    }
+    PhysXShape* wrapperShape = reinterpret_cast<PhysXShape*>(wrapperPtrShape);
+    if (!(filterData.word0 & wrapperShape->getMask()) || !(filterData.word1 & wrapperShape->getGroup())) {
+        return physx::PxQueryHitType::eNONE;
+    }
+    return physx::PxQueryHitType::eBLOCK;
+}
+        
+physx::PxQueryHitType::Enum QueryFilterCallback::postFilter(const physx::PxFilterData& filterData, const physx::PxQueryHit& hit){
+    PX_UNUSED(filterData);
+    PX_UNUSED(hit);
+    return physx::PxQueryHitType::eNONE;
+}
+
+PhysXCharacterController::PhysXCharacterController() {
+    _mObjectID = PhysXWorld::getInstance().addWrapperObject(reinterpret_cast<uintptr_t>(this));
+    _mFilterData = physx::PxFilterData(1, 1, 10000, 0 );
+};
+
+void PhysXCharacterController::release() {
+    eraseFromCCTMap();
+    if(_impl){
+        _impl->release();
+        _impl = nullptr;
+    }
+}
+
+bool PhysXCharacterController::initialize(Node *node) {
+    _mNode = node;
+    onComponentSet();
+
+    if (_impl == nullptr) {
+        return false;
+    } else {
+        PhysXWorld::getInstance().addCCT(*this);
+        return true;
+    }    
+}
+
+void PhysXCharacterController::onEnable() {
+    _mEnabled = true;
+}
+
+void PhysXCharacterController::onDisable() {
+    _mEnabled = false;
+}
+
+void PhysXCharacterController::onDestroy() {
+    release();
+    PhysXWorld::getInstance().removeCCT(*this);
+    PhysXWorld::getInstance().removeWrapperObject(_mObjectID);
+}
+
+cc::Vec3 PhysXCharacterController::getPosition() {
+    const physx::PxExtendedVec3& pos = _impl->getPosition();
+    cc::Vec3 cv(pos.x, pos.y, pos.z);
+    return cv;
+}
+
+void PhysXCharacterController::setPosition(float x, float y, float z) {
+    _impl->setPosition(physx::PxExtendedVec3{x, y, z});
+}
+
+bool PhysXCharacterController::onGround() {
+    return (_pxCollisionFlags & physx::PxControllerCollisionFlag::Enum::eCOLLISION_DOWN);
+}
+
+void PhysXCharacterController::syncSceneToPhysics() {
+    uint32_t getChangedFlags = _mNode->getChangedFlags();
+    if (getChangedFlags & static_cast<uint32_t>(TransformBit::SCALE)) syncScale();
+}
+
+void PhysXCharacterController::syncScale () {
+    updateScale();
+}
+
+//move
+void PhysXCharacterController::move(float x, float y, float z, float minDist, float elapsedTime) {
+    physx::PxVec3 disp{x, y, z};
+    controllerFilter.mFilterData = &_mFilterData;
+    controllerFilter.mFilterCallback = &_mFilterCallback;
+    PhysXWorld::getInstance().getControllerManager().setOverlapRecoveryModule(_mOverlapRecovery);
+    _pxCollisionFlags = _impl->move(disp, minDist, elapsedTime, controllerFilter);
+}
+
+void PhysXCharacterController::setMinMoveDistance(float v) {
+    _mMinMoveDistance = v;
+    //_impl->setMinMoveDistance(v);
+    if(_impl){
+        create();
```

Comment:
Is it necessary to update `minMoveDistance` by re-creating? Recreation is a heavy operation.

Reply:
yes, this is not right.